browserprocess/register.go

```diff
@@ -0,0 +1,59 @@
+package browserprocess
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/grafana/xk6-browser/log"
+)
+
+type processState struct {
+	pid int
+}
+
+var (
+	browserProcessRegister   = map[string]*processState{} //nolint:gochecknoglobals
+	browserProcessRegisterMu = sync.Mutex{}               //nolint:gochecknoglobals
+)
+
+func register(ctx context.Context, logger *log.Logger, pid int) {
+	browserProcessRegisterMu.Lock()
+	defer browserProcessRegisterMu.Unlock()
+
+	iID := GetIterationID(ctx)
+	key := strconv.FormatInt(int64(pid), 10) + iID
+
+	logger.Debugf("BrowserProcess:register", "registered BrowserProcess pid %d", pid)
+
+	browserProcessRegister[key] = &processState{pid: pid}
+}
+
+// ForceProcessShutdown should be called when
+// xk6-browser is having to shutdown due to an
+// internal error (and therefore a panic).
+func ForceProcessShutdown(ctx context.Context) {
+	browserProcessRegisterMu.Lock()
+	defer browserProcessRegisterMu.Unlock()
+
+	iID := GetIterationID(ctx)
+
+	for k, v := range browserProcessRegister {
+		if iID != "" && !strings.Contains(k, iID) {
```

Comment:
I'm a little bit skeptical about generating the keys for the register as a concatenation of `iterationID` and `pid`, and also about using `strings.Contains` to verify. I was experimenting with using to different data structures for the map from `iID->pid` and the list of `pid` but was resulting rather complex, I will try to provide an alternative. Did you consider other options?

Reply:
Yes, it's a smell. I like the approach you suggested 👍